add Flux Administrator's Guide

grondo opened the pull request:
This is a *very* rough draft of a Flux admin doc meant as a reference for sysadmins managing system instances of Flux.

I basically cribbed completely from @garlick's previous work, starting with the outline proposed in #58, and cut-and-pasting a lot of text from the [Small system instance setup](https://github.com/flux-framework/flux-core/wiki/Small-system-instance-setup) wiki page.

I wanted to post this draft so people could comment early if the guide isn't on the right track.

grondo commented:
Hm, not sure I understand why the sphinx build failed:
```
Exception occurred:
  File "/usr/local/lib/python3.8/subprocess.py", line 1702, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'git'
The full traceback has been saved in /tmp/sphinx-err-43sh7w2u.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
cat: _build/output.txt: No such file or directory
make: *** [Makefile:25: spelling] Error 1
```

garlick commented on adminguide.rst:
```diff
@@ -0,0 +1,514 @@
+
+Resources may be temporarily removed from scheduling via the
+``flux resource drain`` command. Currently, resources may only be
+drained at the granularity of rank, for example:
```
Or "Drained at the granularity of a node, represented by its broker rank"?

garlick commented on adminguide.rst:
```diff
@@ -0,0 +1,514 @@
+Managing the Flux Queue
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The queue of jobs is managed by the flux job-manager, which in turn
```
Would it be helpful to at least mention the queue listing command `flux jobs`? (maybe reference its man page?)

garlick commented on adminguide.rst:
```diff
@@ -0,0 +1,514 @@
+In the *idle* state, jobs may still be pending. ``flux queue idle``
+is most useful when the queue is *stopped*.
+
+To query the current status of the queue use the ``flux queue status``
```
Just a thought but maybe we should encourage the use of "idle" for shutting down the system instance, since disallowing new job submissions might cause a running job that submits another job to fail?  

At least it might be helpful to note that use case for jobs, which motivated the `flux queue idle` method of waiting for the compute nodes to become idle.

garlick commented on adminguide.rst:
```diff
@@ -0,0 +1,514 @@
+ [access]
+ allow-guest-user = true
+
+For a system instance it may be convenient to allow the ``root`` user to
+act as the instance owner (so that ``root`` can list and cancel jobs,
+access the KVS, etc). If this is desired, then ``allow-root-owner = true``
+should be set in the ``[access]`` table as well.
```
Could mention that the alternative is to require administrators to run Flux administration commands as the flux user?

cmoussa1 commented:
Looks like @SteVwonder pointed out the same error in #62, in commit 1a7c8c7fd7fb5692403afe48f72a90d011a4077c:

```
 Exception occurred:
  File "/usr/local/lib/python3.8/subprocess.py", line 1702, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
 FileNotFoundError: [Errno 2] No such file or directory: 'git'
```

Maybe the change there will fix the error you are getting?

garlick commented on adminguide.rst:
```diff
@@ -0,0 +1,514 @@
+
+.. _systemd-journal:
+
+~~~~~~~~~~~~~~~
+Systemd journal
+~~~~~~~~~~~~~~~
```
Might mention `journalctl -u flux` as the way to access Flux log messages that the broker emits to stderr?

garlick reviewed (approved):
Thanks for doing this @grondo!
I've added some inline comments and per our discussion offline, I may just force push some of them to this branch this morning.

grondo commented:
> Maybe the change he made there to .github/workflows/main.yml will fix the error you are getting?

Thanks for pointing that out! I've merged  #62 so I'll rebase this one shortly.

grondo commented:
Thanks @garlick! I squashed your work into the initial commit that adds the draft admin guide (with `Co-authored-by`) and forced a push. I also added a note that expedite and hold of jobs are planned features that are not currently supported.

I think I'll go ahead and set MWP now so that we have something to start from on the official docs site.

We can elaborate and edit via future PRs.

garlick commented:
Thanks @grondo for overcoming the activation energy to get something in!
